pkg/controller/discovery/container/collection.go

```diff
@@ -75,7 +75,6 @@ func (r *BaseCollection) IsReady() bool {
 // Reset `hasReconciled` and association with a DataSource.
 func (r *BaseCollection) Reset() {
 	r.hasReconciled = false
-	r.ds = nil
```

Comment:
Predicates may still be using this until the controller (watches) are completely terminated.  Has caused SEGV.
No value in setting to `nil`.

Reply:
I think I just hit this problem in my env, actually.